assets/src/components/map.tsx

```diff
@@ -0,0 +1,131 @@
+import Leaflet, { Map as LeafletMap, Marker } from "leaflet"
+import React, {
+  MutableRefObject,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState,
+} from "react"
+import { ScheduleAdherenceStatus } from "../skate"
+
+interface Props {
+  bearing: number
+  label: string
+  latitude: number
+  longitude: number
+  scheduleAdherenceStatus: ScheduleAdherenceStatus
+}
+
+interface State {
+  map: LeafletMap | null
+  vehicleIcon: Marker | null
+  vehicleLabel: Marker | null
+}
+
+const iconAnchor: [number, number] = [12, 12]
+
+export const updateMap = (
+  { bearing, latitude, longitude, scheduleAdherenceStatus }: Props,
+  {
+    map,
+    vehicleIcon,
+    vehicleLabel,
+  }: {
+    map: LeafletMap
+    vehicleIcon: Marker
+    vehicleLabel: Marker
+  }
+): {
+  map: LeafletMap
+  vehicleIcon: Marker
+  vehicleLabel: Marker
+} => {
+  const zoom = map.getZoom()
+
+  const icon = Leaflet.divIcon({
+    className: `m-vehicle-map__icon ${scheduleAdherenceStatus}`,
+    html: `<svg
+        height="24"
+        viewBox="0 0 24 24"
+        width="24"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <path
+          class="m-vehicle-properties-panel__schedule-adherence-status-icon"
+          d="m10 2.7-6.21 16.94a2.33 2.33 0 0 0 1.38 3 2.36 2.36 0 0 0 1.93-.14l4.9-2.67 4.89 2.71a2.34 2.34 0 0 0 3.34-2.8l-5.81-17a2.34 2.34 0 0 0 -4.4 0z"
+          transform-origin="${iconAnchor[0]}px ${iconAnchor[1]}px"
+          transform="rotate(${bearing})"
+        />
+      </svg>`,
+    iconAnchor,
+  })
+
+  map.setView([latitude, longitude], zoom)
+
+  vehicleIcon.setLatLng([latitude, longitude])
+  vehicleLabel.setLatLng([latitude, longitude])
+
+  vehicleIcon.setIcon(icon)
```

Comment:
If the `label` changes (won't happen as vehicle ids, but could happen if we change the label to represent run numbers or badge numbers or something else), then it'd need to be set like the vehicleIcon is.

I think defining the label svg up here might also make it a little easier to see how the label svg and icon svg relate to each other. (i.e. how they're positioned relative to the latlng)